libs/ui-lib/lib/cim/components/ClusterDeployment/ClusterDetailsFormFields.tsx

```diff
@@ -155,6 +177,26 @@ export const ClusterDetailsFormFields: React.FC<ClusterDetailsFormFieldsProps> =
       {!isNutanix && (
         <CpuArchitectureDropdown cpuArchitectures={cpuArchitectures} isDisabled={isEditFlow} />
       )}
+
+      <CheckboxField
+        name={'addCustomManifests'}
+        label={'Include custom manifests'}
+        helperText={
+          'Additional manifests will be applied at the install time for advanced configuration of the cluster.'
+        }
+        onChange={(val) => {
+          if (!val && customManifests) {
+            setShowModal(true);
+          }
+        }}
+      />
+
+      <DeleteCustomManifestModal
+        isOpen={showModal}
+        onClose={() => setShowModal(false)}
+        onDelete={onDeleteManifests}
+      />
```

Comment:
lets just completely remove the evaluation of this component if it is not open
```suggestion
      {showModal && <DeleteCustomManifestModal
        isOpen
        onClose={() => setShowModal(false)}
        onDelete={onDeleteManifests}
      />}
```